src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/SDKVersionTelemetryServiceComponent.SDKVersionTelemetryServiceInstance.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Telemetry;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies
+{
+    internal partial class SDKVersionTelemetryServiceComponent
+    {
+        private class SDKVersionTelemetryServiceInstance : AbstractProjectDynamicLoadInstance
+        {
+            private const string TelemetryEventName = "SDKVersion";
+            private const string ProjectProperty = "Project";
+            private const string NameProperty = "Name";
+            private const string VersionProperty = "Version";
+
+            private readonly IUnconfiguredProjectVsServices _projectVsServices;
+            private readonly ITelemetryService _telemetryService;
+            private string _projectId;
+
+            [ImportingConstructor]
+            public SDKVersionTelemetryServiceInstance(
+                IUnconfiguredProjectVsServices projectVsServices,
+                ITelemetryService telemetryService)
+                : base(projectVsServices.ThreadingService.JoinableTaskContext)
+            {
+                _projectVsServices = projectVsServices;
+                _telemetryService = telemetryService;
+            }
+
+            protected override async Task InitializeCoreAsync(CancellationToken cancellationToken)
+            {
+                await _projectVsServices.ThreadingService.SwitchToUIThread();
```

Comment:
You are switching to the UI thread, only to immediately schedule on ThreadPool very next line (https://github.com/Microsoft/vs-threading/blob/master/doc/cookbook_vs.md#should-i-await-a-task-with-configureawaitfalse).

Does the telemetry service need UI thread?